src/librustc/ty/sty.rs

```diff
@@ -990,7 +990,7 @@ impl<'a, 'gcx, 'tcx> ParamTy {
 pub struct DebruijnIndex {
     /// We maintain the invariant that this is never 0. So 1 indicates
     /// the innermost binder. To ensure this, create with `DebruijnIndex::new`.
```

Comment:
This comment is wrong now, right? Also, why make the field private, especially now that there's no invariant?

Reply:
Comments repairing is listed in my TODO, it'll be fixed once the key changes works. 
the field is private: new `impl DebruijnIndex` provide enough pub-methods to manipulate it.